src/api/c/graphics_common.cpp

```diff
@@ -178,15 +233,15 @@ void ForgeManager::setWindowChartGrid(const forge::Window* window,
 {
     ChartMapIter iter = mChartMap.find(window);
 
-        if(iter != mChartMap.end()) {
-
-    }
-
     if(iter != mChartMap.end()) {
         // ChartVec found. Clear it.
         // TODO: Should we clear this even if r = old_r and c = old_c?
-        for(int i = 0; i < (int)(iter->second).size(); i++)
-            if((iter->second)[i] != NULL) delete (iter->second)[i];
+        for(int i = 0; i < (int)(iter->second).size(); i++) {
```

Comment:
I think clearing is a fair solution for now. There is no API in forge that would "clear" the properties as you have mentioned - which would probably do the same thing as delete and create anyway. That is, delete all the objects associated with the chart and reset axes limits/titles.